File: src/main/java/boj/problems/step6/No2908.java
package boj.problems.step6;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class No2908 {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter output = new BufferedWriter(new OutputStreamWriter(System.out));

        solve(input, output);

        output.flush();
        output.close();
    }

    static void solve(BufferedReader input, BufferedWriter output) throws IOException {
        StringTokenizer st = new StringTokenizer(input.readLine());

        int a = reverseNumber(st.nextToken());
        int b = reverseNumber(st.nextToken());

        output.write((a > b ? a : b) + "\n");
    }

    private static int reverseNumber(String str) {
        StringBuffer sb = new StringBuffer(str);

        return Integer.parseInt(sb.reverse().toString());
    }
}

File: src/main/kotlin/boj/problems/step6/No2908.kt
package boj.problems.step6

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.IOException
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

object No2908 {
    @Throws(IOException::class)
    fun solve(input: BufferedReader): Int {
        val st = StringTokenizer(input.readLine())
        val a = reverseNumber(st.nextToken())
        val b = reverseNumber(st.nextToken())
        return a.coerceAtLeast(b)
    }

    private fun reverseNumber(str: String): Int {
        val sb = StringBuilder(str)
        return sb.reverse().toString().toInt()
    }
}

fun main() {
    val input = BufferedReader(InputStreamReader(System.`in`))
    val output = BufferedWriter(OutputStreamWriter(System.out))

    output.write(No2908.solve(input).toString())

    input.close()
    output.flush()
    output.close()
}
